packager/media/formats/webm/mkv_writer.cc

```diff
@@ -13,8 +13,18 @@ MkvWriter::MkvWriter() : position_(0) {}
 
```

Comment:
It will be cleaner to implement another buffer based MkvWriter. We may call it `BufferMkvWriter`. We can then switch multi_segment_segmenter to use the new MkvWriter. Optionally we can switch `two_pass_single_segment_segmenter` to use the new MkvWriter too.

Reply:
Makes sense. Let me try it out.